config/config.go

```diff
@@ -68,6 +70,10 @@ type Config struct {
 
 	// Revision contains the current revision of the config.
 	Revision string
+
+	// unprocessedConfig stores the unprocessed version of the config that will be cached.
+	// This version is kept because the config is changed as it moves through the system.
+	unprocessedConfig *Config
```

Comment:
[minor] maybe we can shorten this field name - conceptually an "unprocessed" config is the marshaled config struct, right?

```suggestion
	// raw stores the unprocessed version of the config that will be cached.
	// This version is kept because the config is changed as it moves through the system.
	raw *Config
```

Reply:
changed the name to `toCache` to be clearer, what do you think?

Reply:
yep makes sense